supervision/detection/utils.py

```diff
@@ -48,6 +48,36 @@ def polygon_to_mask(polygon: np.ndarray, resolution_wh: Tuple[int, int]) -> np.n
     return mask
 
 
+def box_iou(box1: np.ndarray, box2: np.ndarray) -> float:
+    """
+    Compute the Intersection over Union (IoU) between two bounding boxes.
+
+    Args:
+        box1 (np.ndarray): A bounding box represented as [x1, y1, x2, y2].
+        box2 (np.ndarray): A bounding box represented as [x1, y1, x2, y2].
```

Comment:
After addressing https://github.com/roboflow/supervision/pull/1874/files#r2202860137, please ensure to update the args description as well.